### ISSUE TITLE
MPM as core app discussion

### ISSUE THREAD

**Issue by RiccardoRossi** [KratosMultiphysics/Kratos#5331]
@KratosMultiphysics/mpm @KratosMultiphysics/structural-mechanics @KratosMultiphysics/technical-committee 

i am opening this issue to centralize the discussion of MPM as core application

**Comment by RiccardoRossi** [comment]
@antonialarese i am writing here my personal opinion about the CL as part of the structural mechanics or as a standalone application.

from my understanding we have (and we should have) essentially 3 possibilities for the constitutive law:

1. small strain, displacement driven:  given a strain vector we obtain a stress
2. large strain, displacement driven: given F (deformation gradient) we obtain S (PK2) or any other strain measure
3. strain-rate driven CLs: given grad_sym_vel we obtain sigma

in all the cases by convention constitutive laws are based on the **TOTAL** strain or F **and NOT on their increment**

In order to have a consisten behaviour (@tteschemacher this is in the direction of your offline comments) we need to guarantee that any constitutive law we add behaves consistently. To my understanding, as a very minimum this implies than any CL should pass a patch test in which multiple loading steps with the same strain level are applied without causing any variation in the stress level (which of course needs to match the analytic value). 

Of course small strain laws should be tested with small strain elements etc, so we would have something like:

* small strain laws -- should be tested against SmallDisplacement elements 
* large strain laws -- should be testes (as a minimum) against Total Lagrangian element
* strain rate (velocity based) laws should be tested against non-newtonian fluid implementations

since the reference SmallDisplacement and TotalLagrangian elements are implemented in the StructuralMechanicsApplication i consider as a requirement that any displacement-driven CL should be guaranteed to pass the (multi step) patch test when using such elements.
Similarly i would say that any velocity driven strain-rate CL should work with fluid elements.

The reason for which i go through such a long preamble is that i am personally NOT against having a separated ConstitutiveLawApplication, **as long as the guarantees i mention are met**.
However providing such guarantees would imply that StructuralMechanics and FluidDynamics would depend on such app and that __any CL__ in the core should be tested against the corresponding reference element before making the cut for being in the core.
The point is that if we want to have such a guarantee (which i consider crucial for consistency) we will need to run the structural mechanics tests (although not recompile the app) every time we want to push new stuff to the CLApp. 

in short what i would not be against having a separate CLapp, provided that:
**any CL in such app is tested against the appropriate element of the StructuralMechanicsApp (or of the FluidDynamics fo the strain driven case)**. [EDIT:] Note that this does not imply that the same CL cannot be tested ALSO in other applications, simply that AS A MINIMUM it needs to be tested in the StructuralMechanics and/or in the FluidDynamics

once against note that this is just my point of view and does not reflect a decision of the @KratosMultiphysics/technical-committee

**Comment by loumalouomega** [comment]
@RiccardoRossi related with the CL, I will send you a private message. It is something good, don't worry

**Comment by antonialarese** [comment]
@RiccardoRossi Your comments are absolutely reasonable. As you said this will apply not only to the StructuralMechApp, but to any other application using some specific CL. Considering the possibly upcoming geomechanics app, this will also apply extensively there (@Vahid-Galavi)

**Comment by Vahid-Galavi** [comment]
I personally prefer a stand alone application for CLs as they should be separated from elements. In case of Geomechanics application, it does not matter to put them in a separate application or in the Structural application as the Geomechanics application is derived from the Structural app. We have to compile the structural app anyway.

**Comment by roigcarlo** [comment]
Closing, as Geo is now completely different and Particle has even changed name.
